reactive/kotlinx-coroutines-rx2/src/RxScheduler.kt

```diff
@@ -5,24 +5,87 @@
 package kotlinx.coroutines.rx2
 
 import io.reactivex.Scheduler
+import io.reactivex.disposables.Disposable
+import io.reactivex.disposables.Disposables
+import io.reactivex.plugins.RxJavaPlugins
 import kotlinx.coroutines.*
 import java.util.concurrent.TimeUnit
 import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
 
 /**
  * Converts an instance of [Scheduler] to an implementation of [CoroutineDispatcher]
  * and provides native support of [delay] and [withTimeout].
  */
 public fun Scheduler.asCoroutineDispatcher(): SchedulerCoroutineDispatcher = SchedulerCoroutineDispatcher(this)
 
+/**
+ * Converts an instance of [CoroutineDispatcher] to an implementation of [Scheduler].
+ */
+public fun CoroutineDispatcher.asScheduler(): Scheduler =
+        if (this is SchedulerCoroutineDispatcher) {
+            scheduler
+        } else {
+            DispatcherScheduler(this)
+        }
+
+private class DispatcherScheduler(private val dispatcher: CoroutineDispatcher) : Scheduler() {
+
+    private val scope = CoroutineScope(SupervisorJob())
```

Comment:
There does not seem to be any need to create a `Job` for this parent scope. It is used to create child worker jobs, but it is not really needed. There's no code that would ever cancel this parent job.

Reply:
@elizarov the parent scope is cancelled in the ```shutdown``` method:

```kotlin
    override fun shutdown() {
        scope.cancel()
    }
```